### PR TITLE
[JN-987] Survey designer plays nicely with internationalized surveys

### PR DESCRIPTION
**Pull request by MatthewBemis**
#### DESCRIPTION

The survey designer will now display the content from the `default` language instead of `[object Object]`.

Other languages cannot be viewed or edited in the designer, but they can be previewed in the "Preview" tab. If you try to edit the value of an element that's internationalized, it will work for the `default` language, but will revert to an un-internationalized version. I think this is okay for now.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

* Repopulate demo
* Load the Basics survey and confirm that you see the actual text values and that there aren't any [object Object]s in the text boxes: https://localhost:3000/demo/studies/heartdemo/env/sandbox/forms/surveys/hd_hd_basicInfo

### CODE REVIEW

File: ui-core/src/types/forms.ts [hunk 1]
```diff
@@ -111,6 +111,19 @@ export type FormContentPage = BaseElement & {
   elements: FormElement[]
 }
 
+/**
+ *  Certain SurveyJS elements can take on multiple forms. For example, the "title" field
+ *  for a question could either be a string, or an object mapping language codes to strings.
+ *  "default" is always present; the other languages are arbitrary.
+ */
+export type I18nMapElement = {
```
Comment (MatthewBemis) [inline ui-core/src/types/forms.ts]: I don't like this name, but I haven't yet decided what I'm going to rename it to

File: ui-admin/src/util/juniperSurveyUtils.tsx [hunk 2]
```diff
@@ -141,3 +142,15 @@ export function questionFromRawText(rawText: string): QuestionObj {
 ReactQuestionFactory.Instance.registerQuestion('addressvalidation', props => {
   return React.createElement(SurveyQuestionAddressValidation, props)
 })
+
+/**
+ * Takes in an I18nSurveyElement and either returns the default value (if I18nSurveyElement is in the
+ * internationalized format), or the string itself (if I18nSurveyElement is a simple string).'
+ * Once the survey designer is fully
+ */
+export const getI18nSurveyElement = (element?: I18nSurveyElement, language?: string) => {
```
Comment (devonbush) [inline ui-admin/src/util/juniperSurveyUtils.tsx]: making the argument default cleans up the ternary below
```suggestion
export const getI18nSurveyElement = (element?: I18nSurveyElement, language = 'default') => {
```

File: ui-core/src/types/forms.ts (hunk 1 above)
Comment (devonbush) [reply ui-core/src/types/forms.ts]: I would just call this `I18nMap`

File: ui-admin/src/util/juniperSurveyUtils.tsx (hunk 2 above)
Comment (devonbush) [reply ui-admin/src/util/juniperSurveyUtils.tsx]: I might also rename this to `i18nSurveyText` to be a bit shorter

**Review by devonbush — APPROVED** [review]
works well locally.  Just some style suggestions

**Review by connorlbark — APPROVED** [review]

**Pull request merged**